src/main/java/ticTacToe/ui/Ui.java

```diff
@@ -24,51 +24,60 @@ public Ui(PrintStream output, InputStream input) {
     }
 
     public void welcomePlayer() {
-        output.println("Welcome to tic-tac-toe");
+        clearScreen();
+        output.println("Welcome to TIC TAC TOE!");
         output.println("Are you ready to play??");
     }
 
     public void printGrid(Lines lines, Grid grid) {
-        output.println(gridPreparer.prepareGridForPrinting(lines, grid));
+        List<ArrayList<String>> rowsWithColours = gridPreparer.rowsWithColours(lines.getRows(grid));
+        output.println(gridPreparer.prepareGridForPrinting(rowsWithColours));
     }
 
     public String chooseOpponent() {
+        output.println("\n");
         output.println("Choose opponent type:");
-        output.println("1- human player");
-        output.println("2- smart computer.");
+        opponentOptions();
         return validOptionNumber(input.nextLine());
     }
 
-    public String askForMarkType() {
-        output.println("First player: please choose a mark: X => Cross, O (letter) => Nought");
+    public Mark askForMarkType() {
+        clearScreen();
+        output.println("First player: please choose a mark: " + GridPreparer.blue_X + " => Cross, " + GridPreparer.red_O + " (letter) => Nought");
         return validMarkType(input.nextLine().toUpperCase());
     }
 
-    public String askForStarter() {
-        output.println("Who starts: X or O (letter)?");
-        return validMarkType(input.nextLine().toUpperCase());
+    public Mark askForStarter() {
+        clearScreen();
+        output.println("Who starts: " + GridPreparer.blue_X + " or " + GridPreparer.red_O + " (letter)?");
+        Mark markType = validMarkType(input.nextLine().toUpperCase());
+        clearScreen();
+        return markType;
     }
 
-    public void promptForPosition(String mark, Lines lines, Grid grid) {
-        printGrid(lines, grid);
-        output.println(String.format("Player %s: please choose a valid position in the grid", mark));
+    public void promptForPosition(Mark mark) {
+        output.println(String.format("Player %s: please choose a valid position in the grid", mark.sign));
     }
 
-    public String validPosition(Grid grid, String mark, Lines lines) {
+    public String validPosition(Grid grid, Mark mark, Lines lines) {
+        printGrid(lines, grid);
         String validatedPosition = validatePosition(grid, mark, lines, validInput(grid, mark, lines, input.nextLine()));
         return notOccupiedPosition(grid, validatedPosition, mark, lines);
     }
 
-    public void confirmMove(String mark, String gridPosition) {
-        output.println(String.format("Player %s marked position %s.", mark, gridPosition));
+    public void confirmMove(Mark mark, String gridPosition) {
+        clearScreen();
+        output.println(String.format("Player %s marked position %s.", mark.sign, gridPosition));
     }
 
-    public void declareWinner(String mark, Lines lines, Grid grid) {
+    public void declareWinner(String  mark, Lines lines, Grid grid) {
```

Comment:
I think the first argument can now be `Mark mark`.

Reply:
Definitely. I missed that out, thanks